tests/simplewaveequation.py

```diff
@@ -0,0 +1,236 @@
+from opesci import *
+from os import path
+from argparse import ArgumentParser, RawTextHelpFormatter
+
+
+_test_dir = path.join(path.dirname(__file__), "src")
+
+
+def simplewave3d(domain_size, grid_size, dt, tmax, output_vts=False, o_converge=True,
+                 accuracy_order=[1, 2, 2, 2],
+                 omp=True, simd=False, ivdep=True, double=False, pluto=False,
+                 filename='test.cpp', expand=True, eval_const=True,
+                 fission=False):
+    """
+    create 3D eigen waves and run FD simulation
+    :param domain_size: define size of domain
+    e.g. (1.0, 1.0, 1.0) for unit cube
+    :param grid_size: define grid size, e.g. (10, 10, 10)
+    :param dt: define time step
+    :param tmax: define simulation time
+    :param output_vts: switch for code to output at each time step
+    (such as save field as vtk file), default False (no output)
+    :param o_converge: switch for code to calculate L2 norms
+    for convergence test, default True (output)
+    :param omp: swtich for inserting #pragma omp for before outer loop
+    default True (use omp)
+    :param simd: switch for inserting #pragma simd before inner loop
+    default False (do not use simd)
+    :param ivdep: switch for inserting #praga ivdep before inner loop
+    default True (use ivdep)
+    default False (not include vtk header files)
+    :param double: switch for using double as real number variables
+    default False (use float)
+    :param expand: expand kernel fully (no factorisation), default True
+    :param eval_const: evaluate all constants in kernel in generated code default True
+    :param filename: output source code file name as string
+    :param pluto: switch for inserting #pragma scop and #pragma endscop for
+    pluto optimisation
+    :param fission: switch for doing loop fission optimisation
+    """
+
+    print 'domain size: ' + str(domain_size)
+    print 'grid size: ' + str(grid_size)
+    print 'approximation order: ' + str(accuracy_order)
+    print 'dt: ' + str(dt)
+    print 'tmax: ' + str(tmax)
+
+    # Declare fields
+    MAIN_GRID = RegularField('MAIN_GRID', dimension=3)
+
+    grid = RegularGrid(dimension=3, domain_size=domain_size, grid_size=grid_size, fields=[MAIN_GRID], pluto=pluto, fission=fission)
+    grid.set_time_step(dt, tmax)
+    grid.set_switches(omp=omp, simd=simd, ivdep=ivdep, double=double,
+                      expand=expand, eval_const=eval_const,
+                      output_vts=output_vts, converge=o_converge)
+    # define parameters
+    # const_c: The constant from the wave equation
+    x, y, z, const_c = symbols('x y z c')
+    grid.set_index([x, y, z])
+    # Set the values of the constant C from the wave equation, and the initial velocity (single scalar value for all dimensions=0)
+    grid.set_params(c=2, v=1)
+
+    # Check if its possible to calculate a limit for the simple case
+    # print 'require dt < ' + str(grid.get_time_step_limit())
```

Comment:
This line is present in eigenwave.py as well. This was commented before because I wanted to confirm how the CFL condition changes for this case. The function has been reimplemented and the line uncommented now.